### PR TITLE
Implemented new features:

### PR DESCRIPTION
**Pull request by kdp-cloud**
- BsaI Digestion
- Filtering out the relevant features
- Written tests to test each case of a BsaI digestion

### CODE REVIEW

File: server/tests/test_server.py [hunk 1]
```diff
@@ -1,5 +1,64 @@
...
+
+def test_sequence_min_length():
+    assert digest_sequence(VectorLevel.LEVEL0, "GGTCTCXXXXXXGAGACC") == (
+        7,
+        7,
+        "",
+    )
+
+
+def test_single_character():
```
Comment (MaybeJustJames) [inline server/tests/test_server.py:33]: Keep this test but name it better. Also re-create a test for a single nucleotide "gene" of interest and call it `test_single_character`.

File: server/tests/test_server.py [hunk 2]
```diff
@@ -1,5 +1,64 @@
+from hmac import digest
```
Comment (MaybeJustJames) [inline server/tests/test_server.py:1]: I don't think you want this. Remove

File: server/tests/test_server.py [hunk 3]
```diff
@@ -1,5 +1,64 @@
...
+    assert digest_sequence(
+        VectorLevel.LEVEL0, "OF_INTERESTXXXXXGAGACCNOT_OF_INTERESTGGTCTCXGENE_"
+    ) == (
+        44,
+        11,
+        "GENE_OF_INTEREST",
+    )
+
+
+def test_normal_backbone():
```
Comment (MaybeJustJames) [inline server/tests/test_server.py:53]: Add `min_length`, `single_character` and `emoji_padding` versions of both these `_backbone` tests please.

**Review by MaybeJustJames — CHANGES_REQUESTED** [review]
Good work @kdp-cloud. Just a few small changes before this can be merged :)

File: server/tests/test_server.py (hunk 3 above)
Comment (kdp-cloud) [reply server/tests/test_server.py]: This is taken care of in the latest commit.

File: server/tests/test_server.py (hunk 2 above)
Comment (kdp-cloud) [reply server/tests/test_server.py]: Done!

File: server/tests/test_server.py (hunk 1 above)
Comment (kdp-cloud) [reply server/tests/test_server.py]: I've renamed it to test test_single_nucleotide_XXX.
XXX = level0 or level0_backbone or level1_backbone

**Comment by MaybeJustJames** [comment]
Just check the names @kdp-cloud. I see several `level0_backbone` now. I don't think that's what you mean

**Comment by kdp-cloud** [comment]
> Just check the names @kdp-cloud. I see several `level0_backbone` now. I don't think that's what you mean

The way you disgest backones used for Level0 elements is different from the way you digest backbones used for Level1 elements.
In the first case, you want to keep the BsaI sites and in the second case, you don't. Hence level0_backbone and level1_backbone. I wanted to do the different tests for both kinds of backbones, like we did for Level 0 elements.

**Comment by MaybeJustJames** [comment]
Ah I see. Thanks

**Review by MaybeJustJames — APPROVED** [review]

**Pull request merged**